src/Avalonia.Documents/Documents/IAddChild.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/***************************************************************************\
+*
+*
+* Description:
+*
+*
+\***************************************************************************/
+
+using System.ComponentModel;
+using System.Windows.Markup;
+using System;
+using MS.Internal;
+
+namespace System.Windows.Markup 
+{
+    ///<summary>
+    /// The IAddChild interface is used for parsing objects that
+    /// allow objects or text underneath their tags in markup that
+    /// do not map directly to a property. 
+    ///</summary>
+    //[Obsolete("IAddChild is obsolete, use the [ContentProperty] attribute instead")]
+    public interface IAddChild
```

Comment:
This interface is redundant. Avalonia's compiler is looking for an `Add` method by default. So every class that previously implemented this interface should expose `void Add(object child)`